views.py

```diff
@@ -749,94 +668,50 @@ def filter(course_id=None):
+    canvas_user_id = request.values.get("custom_canvas_user_id")
+    canvas_domain = request.values.get("custom_canvas_api_domain")
 
     if canvas_domain not in config.ALLOWED_CANVAS_DOMAINS:
         msg = (
-            '<p>This tool is only available from the following domain(s):<br/>{}</p>'
-            '<p>You attempted to access from this domain:<br/>{}</p>'
+            "<p>This tool is only available from the following domain(s):<br/>{}</p>"
+            "<p>You attempted to access from this domain:<br/>{}</p>"
         )
         return render_template(
-            'error.html',
-            message=msg.format(', '.join(config.ALLOWED_CANVAS_DOMAINS), canvas_domain),
+            "error.html",
+            message=msg.format(", ".join(config.ALLOWED_CANVAS_DOMAINS), canvas_domain),
         )
 
-    roles = request.form.get('ext_roles', [])
-    if "Administrator" not in roles and "Instructor" not in roles:
-        return render_template(
-            'error.html',
-            message='Must be an Administrator or Instructor',
-            params=request.form
-        )
+    roles = request.values.get("roles", [])
 
     session["is_admin"] = "Administrator" in roles
+    session["canvas_user_id"] = canvas_user_id
+    session["lti_logged_in"] = True
 
-    key = request.form.get('oauth_consumer_key')
-    if key:
-        secret = oauth_creds.get(key)
-        if secret:
-            tool_provider = ToolProvider(key, secret, request.form)
-        else:
-            tool_provider = ToolProvider(None, None, request.form)
-            tool_provider.lti_msg = 'Your consumer didn\'t use a recognized key'
-            tool_provider.lti_errorlog = 'You did it wrong!'
-            return render_template(
-                'error.html',
-                message='Consumer key wasn\'t recognized',
-                params=request.form
-            )
-    else:
-        return render_template('error.html', message='No consumer key')
-    if not tool_provider.is_valid_request(request):
-        return render_template(
-            'error.html',
-            message='The OAuth signature was invalid',
-            params=request.form
-        )
-
-    if time() - int(tool_provider.oauth_timestamp) > 60 * 60:
-        return render_template('error.html', message='Your request is too old.')
-
-    # This does truly check anything, it's just here to remind you  that real
-    # tools should be checking the OAuth nonce
-    if was_nonce_used_in_last_x_minutes(tool_provider.oauth_nonce, 60):
-        return render_template('error.html', message='Why are you reusing the nonce?')
-
-    session['canvas_user_id'] = canvas_user_id
-    session['lti_logged_in'] = True
-    session['launch_params'] = tool_provider.to_params()
-
-    return redirect(url_for('quiz', course_id=course_id))
+    return redirect(url_for("quiz", course_id=course_id))
 
 
 def was_nonce_used_in_last_x_minutes(nonce, minutes):  # pragma: no cover
```

Comment:
I don't think this is being used anywhere

Reply:
I don't believe it is either.

Reply:
removed.